pkg/activator/net/revision_backends.go

```diff
@@ -89,6 +89,24 @@ const (
 	defaultProbeFrequency time.Duration = 200 * time.Millisecond
 )
 
+// meshMode determines whether we should proxy directly to pods (most efficient), go via the
+// cluster IP (needed when mesh is enabled), or attempt to automatically detect.
+type meshMode int
+
+const (
+	// meshModeAuto first attempts direct pod IP proxying, and then falls back to
+	// cluster IP if this does not work.
+	meshModeAuto meshMode = iota
+	// meshModeEnabled causes us to always use ClusterIP (which means we do not
+	// need to probe, but deactivates activator load balancing)
+	meshModeEnabled
+	// meshModeDisabled causes us to always proxy directly to pods, which will
+	// not work when mesh is enabled, but allows us to intelligently load balance
+	// to pods with capacity, and enables the activator probe optimisation to
+	// start routing to pods before readiness has propagated.
+	meshModeDisabled
+)
```

Comment:
Can we reuse the types from here and the autoscaler?

Reply:
yeah, planning to do this in the next PR when these get added to the context (and therefore the autoscalerconfig package)